change credits icons and tooltips (conditional on feature flag)

christian-byrne opened the pull request:
This PR changes the credits icons and tooltips based on state of the `subscription_tiers_enabled` feature flag.

When the flag is enabled (or undefined -- for local), the dollar icon is replaced with the lucide-component icon in UserCredit and node price badges (Partner Nodes), and a new tooltip row appears in CurrentUserPopover displaying "Credits have been unified" with a detailed hover tooltip explaining the credit unification across Partner Nodes and Cloud workflows.

<img width="539" height="535" alt="image" src="https://github.com/user-attachments/assets/7e952f9b-0abb-4979-85b7-0eecdeaf808c" />

Related:

- https://github.com/Comfy-Org/ComfyUI_frontend/pull/6115 (borrows badge implementation from this PR)

┆Issue is synchronized with this [Notion page](https://www.notion.so/PR-7276-change-credits-icons-and-tooltips-conditional-on-feature-flag-2c46d73d365081809a6afd5861018a15) by [Unito](https://www.unito.io)

christian-byrne committed:
feat: Add subscription tiers UI changes with feature flag toggle

- Add subscription_tiers_enabled feature flag to control new UI elements
- Change dollar icon to lucide-component icon in UserCredit when flag enabled
- Add "Credits have been unified" tooltip row in CurrentUserPopover when flag enabled
- Update node price badges to use lucide-component SVG icon when flag enabled
- Extend LGraphIcon system to support both Unicode fonts and SVG images via ctx.drawImage()
- Maintain backward compatibility - all existing behavior preserved when flag disabled

christian-byrne committed:
feat: Add tooltip to credits unified help icon

- Add detailed tooltip explaining credit unification
- Include information about Partner Nodes and Cloud workflows
- Mention existing balance conversion
- Add proper hover states and delays for better UX

christian-byrne committed:
refactor: Remove unnecessary code comments

Remove obvious comments that don't add value:
- 'Create SVG data URI for lucide-component icon'
- 'SVG image rendering' / 'Font-based Unicode rendering'
- 'Draw icon background circle if bgColor is set'
- 'Draw SVG image' / 'Draw icon'
- Unicode constant inline comments

Code should be self-documenting without redundant explanations.

christian-byrne committed:
fix: Make unified credits text smaller to prevent layout issues

Change text size from text-sm to text-xs to prevent the message from
wrapping and making the top-up button layout awkward.

christian-byrne committed:
fix: Align unified credits text with other elements

Remove px-4 padding from the tooltip row to align with UserCredit.
The Partner Nodes button keeps its pl-6 indentation as intended.

christian-byrne committed:
fix: Make help icon same size as text

Add text-xs class to the question mark icon to match the
unified credits message text size for better visual consistency.

christian-byrne committed:
fix: Restore canvas fillStyle after background circle drawing

Save and restore ctx.fillStyle in the image rendering branch to match
the pattern used in the unicode rendering branch. This prevents the
background circle color from affecting subsequent canvas operations.

AustinMroz commented on src/composables/node/usePriceBadge.ts:
```diff
@@ -1,10 +1,16 @@
 
+const componentIconSvg = new Image()
+componentIconSvg.src =
+  "data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' width='24' height='24' viewBox='0 0 24 24'%3E%3Cpath fill='none' stroke='currentColor' stroke-linecap='round' stroke-linejoin='round' stroke-width='2' d='M15.536 11.293a1 1 0 0 0 0 1.414l2.376 2.377a1 1 0 0 0 1.414 0l2.377-2.377a1 1 0 0 0 0-1.414l-2.377-2.377a1 1 0 0 0-1.414 0zm-13.239 0a1 1 0 0 0 0 1.414l2.377 2.377a1 1 0 0 0 1.414  ...
```
```suggestion
  "data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' width='24' height='24' viewBox='0 0 24 24'%3E%3Cpath fill='none' stroke='oklch(0.828 0.189 84.429)' stroke-linecap='round' stroke-linejoin='round' stroke-width='2' d='M15.536 11.293a1 1 0 0 0 0 1.414l2.376 2.377a1 1 0 0 0 1.414 0l2.377-2.377a1 1 0 0 0 0-1.414l-2.377-2.377a1 1 0 0 0-1.414 0zm-13.239 0a1 1 0 0 0 0 1.414l2.377 2.377a1 1 0 0 0 1.414 0l2.377-2.377a1 1 0 0 0 0-1.414L6.088 8.916a1 1 0 0 0-1.414 0zm6.619 6.619a1 1 0 0 0 0 1.415l2.377 2.376a1 1 0 0 0 1.414 0l2.377-2.376a1 1 0 0 0 0-1.415l-2.377-2.376a1 1 0 0 0-1.414 0zm0-13.238a1 1 0 0 0 0 1.414l2.377 2.376a1 1 0 0 0 1.414 0l2.377-2.376a1 1 0 0 0 0-1.414l-2.377-2.377a1 1 0 0 0-1.414 0z'/%3E%3C/svg%3E"
```

I tried a bunch of different things. Setting ctx styles, hex codes, any combination of properties, and `var(--color-amber-400)` didn't seem to work, but this, appallingly, does.

Wasn't familiar with `currentColor`. It'd be super cool if it actually worked.

christian-byrne committed:
fix: Update partner node icons for both canvas and Vue nodes

- Use #FABC25 stroke color for canvas badges without circular background
- Keep existing #8D6932 badge background color
- Add feature flag toggle to Vue NodeHeader.vue for API node icons
- Switch between dollar-sign and component icons based on subscription_tiers_enabled flag
- Ensure consistent icon behavior across both litegraph and Vue node systems

christian-byrne committed:
fix: Use nullish coalescing in LGraphIcon and add remote config support for subscription tiers

- Replace logical OR with nullish coalescing in LGraphIcon.ts to preserve explicit 0 values
- Update subscriptionTiersEnabled to check remoteConfig first, matching other feature flags pattern
- Add subscription_tiers_enabled to RemoteConfig type definition
